### ISSUE TITLE
Slow filtering on large datasets due to clone

### ISSUE THREAD

**Issue by igor-d-n** [matfish2/vue-tables-2#931]
- steps for reproducing the issue:
just use it with more than 10 rows

Why you use clone at every filtering tick? It's slow as hell. Just removing it giving x100-200 times speed! Not 10-20% it is 20 000%

What will I broke if I change this lines in filtered-data.js:
var data = clone(this.tableData);
to
var data = this.tableData;
and 
this.allFilteredData = JSON.parse(JSON.stringify(data));
to
this.allFilteredData = data;

**Comment by matfish2** [comment]
a. "What will I broke if I change this lines" - Reason for cloning is not to affect the original data set. If I didn't clone, `data` would point to original data and modify it.
b. Ran some tests using performance.now().  `JSON.parse(JSON.stringify(data))` does NOT significantly slow down the filter. However, using `clone` does slow it down very much like you pointed out. I would replace it with `JSON.parse(JSON.stringify(data))` but that would not retain any complex object types such as `moment` dates, which will affect the presentation, and sorting. I'll try to think of a better way. I'm open to suggestions.
